plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/LicenseSuggest.hs

```diff
@@ -52,22 +56,25 @@ licenseErrorAction uri diag =
         edit  = WorkspaceEdit (Just $ Map.singleton uri $ List tedit) Nothing Nothing
       in CodeAction title (Just CodeActionQuickFix) (Just $ List []) Nothing Nothing (Just edit) Nothing Nothing
 
--- | Given an error message returned by 'Ide.Plugin.Cabal.Diag.errorDiagnostic',
+-- | License name of every license supported by cabal
+licenseNames :: [T.Text]
+licenseNames = map (T.pack . licenseId) [minBound .. maxBound]
+
+-- | Given a diagnostic returned by 'Ide.Plugin.Cabal.Diag.errorDiagnostic',
 --   if it represents an "Unknown SPDX license identifier"-error along
 --   with a suggestion then return the suggestion (after the "Do you mean"-text)
 --   along with the incorrect identifier.
-licenseErrorSuggestion
-  :: T.Text
+licenseErrorSuggestion ::
+  T.Text
   -- ^ Output of 'Ide.Plugin.Cabal.Diag.errorDiagnostic'
-  -> Maybe (T.Text, T.Text)
+  -> [(T.Text, T.Text)]
   -- ^ (Original (incorrect) license identifier, suggested replacement)
-licenseErrorSuggestion message =
-  mSuggestion message >>= \case
-    [original, suggestion] -> Just (original, suggestion)
-    _                      -> Nothing
+licenseErrorSuggestion msg = take 10 $
```

Comment:
Should we use our fuzzy matching module that we use elsewhere? I think it's overkill in some senses since we don't have a big list of completions here, but I think it's somewhat nice to use the same mechanism everywhere?

Reply:
What fuzzy matching module do we have? This PR uses `simpleFilter` to fuzzily find best matches in the list of SPDX identifiers.

Reply:
Oh right. We have `Text.Fuzzy.Parallel`, but I guess that's only supposed to be a faster version of `Text.Fuzzy`. Never mind.

Reply:
I did actually opt to `Text.Fuzzy.Parallel`, looks a bit cleaner, thank you for your suggestion!